test/src/main/java/org/axonframework/test/deadline/ScheduledDeadlineInfo.java

```diff
@@ -36,36 +36,52 @@
     private final String deadlineName;
     private final String scheduleId;
     private final int counter;
-    private final Object messageOrPayload;
+    private final DeadlineMessage<?> deadlineMessage;
     private final ScopeDescriptor deadlineScope;
 
     /**
      * Instantiates a ScheduledDeadlineInfo.
      *
-     * @param scheduleTime     The time as an {@link Instant} at which the deadline is scheduled
-     * @param deadlineName     A {@link String} denoting the name of the deadline; can be used together with the
-     *                         {@code scheduleId} to cancel the deadline
-     * @param scheduleId       A {@link String} identifier representing the scheduled deadline; can be used together
-     *                         with the {@code deadlineName} to cancel the deadline
-     * @param counter          Used to differentiate two deadlines scheduled at the same time
-     * @param messageOrPayload The payload of the scheduled deadline. Might be {@code null} if none is provided
-     * @param deadlineScope    A description of the {@link org.axonframework.messaging.Scope} in which the deadline is
-     *                         scheduled
+     * @param scheduleTime    The time as an {@link Instant} at which the deadline is scheduled
+     * @param deadlineName    A {@link String} denoting the name of the deadline; can be used together with the
+     *                        {@code scheduleId} to cancel the deadline
+     * @param scheduleId      A {@link String} identifier representing the scheduled deadline; can be used together
+     *                        with the {@code deadlineName} to cancel the deadline
+     * @param counter         Used to differentiate two deadlines scheduled at the same time
+     * @param deadlineMessage The deadline message of the scheduled deadline.
+     * @param deadlineScope   A description of the {@link org.axonframework.messaging.Scope} in which the deadline is
+     *                        scheduled
      */
     public ScheduledDeadlineInfo(Instant scheduleTime,
                                  String deadlineName,
                                  String scheduleId,
                                  int counter,
-                                 Object messageOrPayload,
+                                 DeadlineMessage<?> deadlineMessage,
                                  ScopeDescriptor deadlineScope) {
         this.scheduleTime = scheduleTime;
         this.deadlineName = deadlineName;
         this.scheduleId = scheduleId;
         this.counter = counter;
-        this.messageOrPayload = messageOrPayload;
+        this.deadlineMessage = deadlineMessage;
         this.deadlineScope = deadlineScope;
     }
 
+    /**
+     * Creates a new instance of scheduled deadline info with all given {@code deadlineMessage}. Other fields are the
+     * same.
+     *
+     * @param deadlineMessage New deadline message
```

Comment:
This line is a bit vague. Not sure what the intended description is.

Reply:
Fixed the line above. Purpose of this parameter is going to be clearer.